Quality_Declaration.md

```diff
@@ -0,0 +1,179 @@
+* [windows_release](https://ci.ros2.org/view/nightly/job/nightly_win_rel/lastBuild/testReport/spdlog_vendor/)
+
+## Dependencies [5]
+
+### Direct and Optional Runtime ROS Dependencies [5.i]/[5.ii]
+
+`spdlog_vendor` does not have direct/optional runtime ROS dependencies.
+
+### Direct Runtime non-ROS Dependency [5.iii]
+
+`spdlog_vendor` depends directly on the external dependency `spdlog`, which is qualified as quality level 4 in its [Quality Declaration](./spdlog_Q_DECLARATION.md).
+
+## Platform Support [6]
+
+`spdlog_vendor` supports all of the tier 1 platforms as described in [REP-2000](https://www.ros.org/reps/rep-2000.html#support-tiers).
+
+## Security [7]
+
+### Vulnerability Disclosure Policy [7.i]
+
+This package does not yet have a Vulnerability Disclosure Policy.
+
+# Current status Summary
+
+The chart below compares the requirements in the REP-2004 with the current state of the spdlog package.
+|Number|  Requirement| Current state |
+|--|--|--|
+|1| **Version policy** |---|
+|1.i|Version Policy available | ✓ |
+|1.ii|Stable version |✓|
+|1.iii|Declared public API|✓|
+|1.iv|API stability policy|✓|
+|1.v|ABI stability policy|✓|
+|1.vi_|API/ABI stable within ros distribution|✓|
+|2| **Change control process** |---|
+|2.i| All changes occur on change request | ✓|
+|2.ii| Contributor origin (DCO, CLA, etc) | ✓|
+|2.iii| Peer review policy | ✓ |
+|2.iv| CI policy for change requests | ✓ |
+|2.v| Documentation policy for change requests | ✓ |
+|3| **Documentation** | --- |
+|3.i| Per feature documentation | ✓ |
+|3.ii| Per public API item documentation | ✓ |
+|3.iii| Declared License(s) | ✓ |
+|3.iv| Copyright in source files| ✓ |
+|3.v.a| Quality declaration linked to README | ✓ |
+|3.v.b| Centralized declaration available for peer review |✓|
+|4| Testing | --- |
+|4.i| Feature items tests | ✓ |
+|4.ii| Public API tests | ✓ |
+|4.iii.a| Using coverage |✓ |
+|4.iii.a| Coverage policy | ✓ |
+|4.iv.a| Performance tests (if applicable) | ✓ |
+|4.iv.b| Performance tests policy| ✓ |
+|4.v.a| Code style enforcement (linters)| ✓ |
+|4.v.b| Use of static analysis tools | ✓ |
+|5| Dependencies | --- |
+|5.i| Must not have ROS lower level dependencies | ✓ |
+|5.ii| Optional ROS lower level dependencies| ✓ |
+|5.iii| Justifies quality use of non-ROS dependencies |✓|
+|6| Platform support | --- |
+|6.i| Support targets Tier1 ROS platforms| ✓ |
+|7| Security | --- |
+|7.i| Vulnerability Disclosure Policy | ☓ |
+
+Comparing this table with the [Quality Level Comparison Chart of REP2004](https://github.com/ros-infrastructure/rep/blob/master/rep-2004.rst#quality-level-comparison-chart) lead us to decide that this package qualifies to Quality Level 4.
+
+To reach Quality level 3, adding a Vulnerability Disclosure will be needed.
+To reach Quality level 2, adding features documentation will be needed.
+
+non-ROS dependencies Quality level will need to be adjusted accordignly to be equal or below to claims made in this Quality Declaration.
```

Comment:
```suggestion
non-ROS dependencies Quality level will need to be adjusted accordingly to be equal or below to claims made in this Quality Declaration.
```

What are you trying to say here?

Reply:
I was trying to imply that to have upper levels on this QD, the `spdlog` external dependency have to increase its QL as well as an extra thought

Reply:
How about:

"spdlog_vendor is limited by the quality level of the upstream dependency spdlog. In order to reach higher quality levels, the issues limiting spdlog will either need to be addressed by spdlog itself, or through adequate quality control measures in this package."